course-material/intro-to-sql/joining-data-in-sql.md

```diff
@@ -1,14 +1,397 @@
+
+### Right Join
+
+Right join is identical to the nature of the left join. A right join will guarantee the inclusion of every row from the "right table", regardless if the key being joined on appears in the "left table."
+
+Here we replace the `LEFT JOIN` in our last example with `RIGHT JOIN`. 
+
+```{code-cell} ipython3
+%%sql 
+SELECT COUNT(*)
+FROM s1.account
+RIGHT JOIN s1.district 
+    ON s1.account.district_id = s1.district.district_id
+```
+
+The resulting `COUNT(*)` of our join omits the one row we `INSERT` into `s1.account` previously. Let's double check and see if our inserted row exists after this join. 
+
+```{code-cell} ipython3
+%%sql 
+SELECT *
+FROM s1.account
+RIGHT JOIN s1.district 
+    ON s1.account.district_id = s1.district.district_id
+WHERE s1.account.district_id = 99999
+```
+
+Exactly what we expect.
+
+What would happen if we reverse the placement of our tables in our `RIGHT JOIN` clause? The results of reversing the table placements are below.
+
+```{code-cell} ipython3
+%%sql 
+SELECT COUNT(*)
+FROM s1.district
+RIGHT JOIN s1.account 
+    ON s1.account.district_id = s1.district.district_id
+```
+
+```{code-cell} ipython3
+%%sql 
+SELECT *
+FROM s1.district
+RIGHT JOIN s1.account 
+    ON s1.account.district_id = s1.district.district_id
+WHERE s1.account.district_id = 99999
+```
+
+We get near exact results from our previous `LEFT JOIN` demonstration. The only difference is the order of our columns in the query output. The `s1.account` table appears on the far right whil the `s1.district` table appears on the left. To ensure complete visibility of the output, please utilize the scroll bar.
+
+### Full Join
+Full join (also known as outer join) results in the inclusion of all rows from both tables. To showcase the full capability of full joins, we first `INSERT` another row into the `s1.district` table with a "district_id" value not present in the `s1.account` table. In other words, we are creating new value for `s1.district`'s primary key that does not appear in `s1.account`'s foreign key. 
+
+```{code-cell} ipython3
+%%sql 
+INSERT INTO s1.district
+VALUES (3333, 'Hypothetical District', 'Hypothetical Region',1,1,1,1,1,1,1,1,1,1,1,1,1)
+```
+
+```{code-cell} ipython3
+%%sql 
+SELECT COUNT(*)
+FROM s1.account 
+FULL JOIN s1.district 
+    ON s1.account.district_id = s1.district.district_id
+```
+
+```{code-cell} ipython3
+%%sql 
+SELECT *
+FROM s1.account
+FULL JOIN s1.district 
+    ON s1.account.district_id = s1.district.district_id
+WHERE s1.account.district_id = 99999 OR s1.district.district_id = 3333
+```
+
+The full join on `s1.district` and `s1.account` results in a query that includes all rows from both tables. The `OR` clause in the last SQL statement verifies this by querying the hypothetical rows we created in each table.
+
+## You try: Use JupySQL to perform the queries and answer the questions.
+
+### Question 1 (Easy):
```

Comment:
There is a lot of background knowledge covered in each section. 

Instead of having them at the end,I'd suggest adding one question per section.